src/app/onboarding/availability/page.tsx

```diff
@@ -1,109 +1,213 @@
+  const isEmpty = useMemo(
     () =>
-      startDate && !isValidDate(startDate)
-        ? 'Must select a current or future date'
-        : '',
-    [startDate],
+      formValues.hoursPerMonth === undefined ||
+      formValues.startDate === undefined,
+    [formValues],
   );
 
+  // update form submitter and dirty state
+  const { setForm: setOnboardingForm } = onboarding;
+  const { isDirty, isValid } = form.formState;
   useEffect(() => {
-    const hpm = onboarding?.userProfile.hours_per_month;
-    setHours(hpm !== undefined && hpm >= 0 ? hpm.toString() : '');
-    setStartDate(onboarding?.userProfile.start_date || '');
-    setPeriods(onboarding?.userProfile.availability_description || '');
-
-    if (
-      hpm !== undefined &&
-      hpm >= 0 &&
-      onboarding?.userProfile.start_date &&
-      isValidDate(onboarding?.userProfile.start_date)
-    ) {
-      onboarding?.setCanContinue(true);
-    } else {
-      onboarding?.setProgress(2);
-      onboarding?.setCanContinue(false);
-    }
-  }, [onboarding]);
-
-  const handleHours = (v: string) => {
-    const n = parseInt(v, 10);
-    if (Number.isNaN(n)) {
-      onboarding?.updateProfile({
-        hours_per_month: undefined,
-      });
-    } else {
-      onboarding?.updateProfile({
-        hours_per_month: n,
-      });
-    }
-  };
-
-  const handleStartDate = (v: string) => {
-    if (v === '' || isValidDate(v)) {
-      onboarding?.updateProfile({
-        start_date: v,
-      });
-    }
-  };
-
-  const handleAvailability = (v: string) => {
-    onboarding?.updateProfile({
-      availability_description: v,
+    setOnboardingForm({
+      trigger: form.handleSubmit(identity),
+      isDirty,
+      isValid,
     });
+  }, [setOnboardingForm, form, isDirty, isValid]);
+
+  // handle valid form submission
+  // validity should have already been handled by Zod
+  const onValidSubmit = () => {
+    push(`/onboarding/${onboarding.flow[pageProgress + 1].url}`);
+    onboarding.setForm(undefined);
   };
 
   return (
-    <>
-      <H1>Availability</H1>
-      <TextInput
-        label="How much time do you have to commit? (hrs/month)"
-        placeholder="hours/month"
-        errorText={getHoursErrorText}
-        type="number"
-        id="hours"
-        value={hours}
-        setValue={setHours}
-        onChange={handleHours}
-      />
-      <DateInput
-        label="What is the earliest you are available to volunteer?"
-        id="start_date"
-        error={getStartDateErrorText}
-        value={startDate}
-        min={parseDateAlt(new Date())}
-        setValue={setStartDate}
-        onChange={handleStartDate}
-      />
-      <TextAreaInput
-        label="Are there specific time periods you will not be available? (Optional)"
-        placeholder="I won't be available from..."
-        id="periods"
-        value={periods}
-        setValue={setPeriods}
-        onChange={handleAvailability}
-      />
-    </>
+    <FormProvider {...form}>
```

Comment:
this is a point for all pages of the flow -- can we autoscroll to the top on screen render? because the submit button is often near the bottom, and so we start each step kinda at the bottom